### DISCUSSION TITLE
Receiving an error in a function related to CTkCanvas and there seems to be something off about it

### DISCUSSION THREAD

**Discussion by shasankp000** [TomSchimansky/CustomTkinter#1014] [Q&A] [answered]
I have a rectangle in a canvas (CTkCanvas) and I have attached two functions on it's "Enter" and "Leave" events which will change the border color of the rectangle to yellow when entered, and set it back to transparent when left.

This is my code : 

``` 
    def change_outline_color_enter(self):
        self.host_canvas.itemconfigure(self.rect, outline="yellow")

  
    def change_outline_color_leave(self):
        self.host_canvas.itemconfigure(self.rect, outline="transparent")

    def create_rectangle(self):

        # 
        self.rect = self.host_canvas.create_rectangle(100, 100, 300, 250, width=5)

        self.host_canvas.tag_bind(self.rect, "<Enter>", self.change_outline_color_enter)
        self.host_canvas.tag_bind(self.rect, "<Leave>", self.change_outline_color_leave)
    
```

The code for the canvas is located elsewhere but I can access it since it is inside a class.

When I run the program I get the error : 

``` 
TypeError: MineCrust.change_outline_color_enter() takes 1 positional argument but 2 were given
Exception in Tkinter callback
Traceback (most recent call last):
  File "C:\Users\shasa\AppData\Local\Programs\Python\Python310\lib\tkinter\__init__.py", line 1921, in __call__
    return self.func(*args)
TypeError: MineCrust.change_outline_color_leave() takes 1 positional argument but 2 were given
```

I believe it's because of the self parameter in the two functions. But if I remove them I won't be able to access the rectangle contained in the create_rectangle function. What should I do? Also the canvas looks odd with the border around it : 
![image](https://user-images.githubusercontent.com/46317225/211198976-799dbd99-6953-4cc8-9ca2-af5455be9f2a.png)

With the white border around it. Can anyone tell me why this is happening and what can I do to remove the white border as well?

**Comment by avalon60** [comment, marked as answer]
I suspect that if you define an "event" parameter in the functions that you are binding, that this will solve the problem.

Eaxmple:

```
    def _copy_harmony_input_colour(self, event=None, shade_copy=False):
        colour = self._btn_keystone_colour.cget('bg')
```

Obviously, you also need to include your own, additional parameters.

**Reply by shasankp000** [reply]
Yea that worked. Thanks. But what about the unusual white border of the canvas? Would you happen to know why is it like that and maybe suggest a solution to it?

**Reply by avalon60** [reply]
Sorry, but not sure - there is little to go on code-wise, and to be perfectly honest, I haven't really played around with canvases.

**Reply by shasankp000** [reply]
Ah, np then, I will scout for more solutions on this then
